code/Calendar.php

```diff
@@ -1015,6 +1016,10 @@ public static function getSortedWeekdays($namesWidth = false, $locale = '')
      */
     public static function getDateFormat($width, $locale = '')
     {
+        if (substr($width, 0, 1) === '~') {
```

Comment:
Fixed. FWIW, I could reproduce the performance issue in PHP 5 and 7.0, but in PHP 7.1 there does not seem to be any performance difference.